tests/Doctrine/Tests/ORM/Functional/MergeSplHashConflictTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\ORM\Query;
+use Doctrine\Tests\Models\CMS\CmsUser;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class MergeSplHashConflictTest extends OrmFunctionalTestCase
+{
+    protected function setUp()
+    {
+        $this->useModelSet('cms');
+        parent::setUp();
+    }
+
+    /**
+     * @test
+     */
+    public function testMergeSplHashConflict()
+    {
+        // Create
+        $user = new CmsUser;
+        $hash = spl_object_hash($user);
+
+        $mergedUser = $this->_em->merge($user);
```

Comment:
We couldn't assertSame as for object assertSame check if the two variables reference the same object. In the case of merge, it will return a copy so it's another object